tests/lib/modem_trace/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(modem_trace_test)
+
+# create mock
+cmock_handle(${ZEPHYR_HAL_NORDIC_MODULE_DIR}/nrfx/drivers/include/nrfx_uarte.h )
+cmock_handle(${ZEPHYR_SEGGER_MODULE_DIR}/SEGGER/SEGGER_RTT.h )
+
+# generate runner for the test
+test_runner_generate(src/modem_trace_test.c)
+
+target_include_directories(app PRIVATE src)
+
+# add test file
+target_sources(app PRIVATE src/modem_trace_test.c)
+
+# add unit under test
+target_sources(app PRIVATE ${NRF_DIR}/lib/nrf_modem_lib/modem_trace.c)
+
+# include paths
+target_include_directories(app PRIVATE ${ZEPHYR_HAL_NORDIC_MODULE_DIR}/nrfx)
+target_include_directories(app PRIVATE ${ZEPHYR_HAL_NORDIC_MODULE_DIR}/nrfx/mdk)
+target_include_directories(app PRIVATE ${ZEPHYR_CMSIS_MODULE_DIR}/CMSIS/Core/Include/)
+target_include_directories(app PRIVATE ${ZEPHYR_SEGGER_MODULE_DIR}/Config/)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/modules/hal_nordic/nrfx)
+target_include_directories(app PRIVATE ${NRF_DIR}/include/modem/)
+target_include_directories(app PRIVATE ${ZEPHYR_NRFXLIB_MODULE_DIR}/nrf_modem/include/ )
+
+# Define the device. Otherwise an "unknown device" warning will be triggered when compiling
+# nrfx_config.h
+target_compile_definitions(app PRIVATE NRF9160_XXAA)
+# Define NRFX_DECLARE_ONLY to simplify generation of mocks. If this is not done, then the
+# the compiler would be unable to compile the generated mocks for nrfx uart driver.
+target_compile_definitions(app PRIVATE NRFX_DECLARE_ONLY)
+# Since Device Tree is not available when running unit tests, we need to provide the hardware
+# details that otherwise would be fetched by the UUT from the Device Tree.
+target_compile_definitions(app PRIVATE DT_N_S_uart_1_P_tx_pin=17)
+target_compile_definitions(app PRIVATE DT_N_S_uart_1_P_rx_pin=18)
+
+if(BOARD STREQUAL qemu_cortex_m3)
+    target_compile_definitions(app PRIVATE DT_N_S_soc_S_uart_4000d000=DT_N_S_uart_1)
+endif()
+
+target_compile_definitions(app PRIVATE NRFX_UARTE1_ENABLED)
+# Define the Up and Down buffer count needed for compiling the Segger RTT headers.
+target_compile_definitions(app PRIVATE
+        CONFIG_SEGGER_RTT_MAX_NUM_UP_BUFFERS=3
+        CONFIG_SEGGER_RTT_MAX_NUM_DOWN_BUFFERS=3)
+
+target_compile_definitions(app PRIVATE CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT_BUF_SIZE=255)
+
+# Preinclude file to the module under test to redefine IS_ENABLED() macro
+# which is used in the module.
+set_property(SOURCE ${NRF_DIR}/lib/nrf_modem_lib/modem_trace.c PROPERTY COMPILE_FLAGS
```

Comment:
why using this instead of target_include_directories?

Reply:
I am making the compiler include the `modem_trace_test.h` when compiling `modem_trace.c`. This way we dont pollute the DUT's source code.

This is also found in the example unit test.

https://github.com/nrfconnect/sdk-nrf/blob/9c087491cc18fe9cd2de6bb7adc9b50b06b1f7d4/tests/unity/example_test/CMakeLists.txt#L34